parachain/runtime/runtime-common/src/lib.rs

```diff
@@ -98,32 +112,46 @@ impl<Balance, AccountId, FeeAssetLocation, EthereumNetwork, AssetTransactor, Fee
 				None
 			})
 			.next();
-
-		if let Some((fee_index, total_fee)) = maybe_total_supplied_fee {
-			let remote_fee = total_fee.saturating_sub(FeeProvider::local_fee());
-			if remote_fee > (0u128).into() {
-				// Refund remote component of fee to physical origin
-				deposit_or_burn_fee::<AssetTransactor, _>(
-					MultiAsset { id: Concrete(token_location), fun: Fungible(remote_fee.into()) }
-						.into(),
-					context,
-					para_sovereign,
-				);
-				// Return remaining fee to the next fee handler in the chain.
-				let mut modified_fees = fees.inner().clone();
-				modified_fees.remove(fee_index);
-				modified_fees.push(MultiAsset {
-					id: Concrete(token_location),
-					fun: Fungible((total_fee - remote_fee).into()),
-				});
-				return modified_fees.into()
-			}
+		if maybe_total_supplied_fee.is_none() {
+			log::error!(
+				target: LOG_TARGET,
+				"could not find fee asset item in fees: {:?}",
+				fees,
+			);
+			return fees
 		}
-
-		log::info!(
-			target: "xcm::fees",
-			"XcmExportFeeToSibling skipped: {fees:?}, context: {context:?}, reason: {reason:?}",
+		let (fee_index, total_fee) = maybe_total_supplied_fee.unwrap();
+		let local_fee = FeeProvider::local_fee();
+		let remote_fee = total_fee.saturating_sub(local_fee);
+		if local_fee == Balance::zero() || remote_fee == Balance::zero() {
+			log::error!(
+				target: LOG_TARGET,
+				"calculated refund incorrect with local_fee: {:?} and remote_fee: {:?}",
+				local_fee,
+				remote_fee,
+			);
+			return fees
+		}
+		// Refund remote component of fee to physical origin
+		let result = AssetTransactor::deposit_asset(
+			&MultiAsset { id: Concrete(token_location), fun: Fungible(remote_fee.into()) },
+			&MultiLocation { parents: 1, interior: X1(Parachain(para_id)) },
+			context,
 		);
-		fees
+		if result.is_err() {
+			log::error!(
+				target: LOG_TARGET,
+				"transact fee asset failed: {:?}",
+				result.unwrap_err()
+			);
+			return fees
+		}
+
+		// Return remaining fee to the next fee handler in the chain.
+		let mut modified_fees = fees.inner().clone();
+		modified_fees.remove(fee_index);
+		modified_fees
+			.push(MultiAsset { id: Concrete(token_location), fun: Fungible(local_fee.into()) });
```

Comment:
Why is the local fee passed on to the next fee handler?

Reply:
Only the remote fee (fee for execution on the destination) needs to be siphoned off to the parachain sovereign. The local fee (the fee charged for this xcm being processed on bridgehub) needs to go to wherever the default fees for xcm go, i.e. the next fee handler. Probably waivered because the origin is AssetHub which is a system parachain.